src/test/java/io/confluent/examples/streams/ReduceTest.java

```diff
@@ -0,0 +1,112 @@
+
+/**
+ * How to aggregate messages via `groupByKey()` and `reduce()` to implement concatenation of data.
+ *
+ * This example can be adapted to structured (nested) data formats such as Avro or JSON in case you need to concatenate
+ * only certain field(s) in the input.
+ *
+ * Note: This example uses lambda expressions and thus works with Java 8+ only.
+ *
+ * See {@link ReduceScalaTest} for the equivalent Scala example.
+ */
+public class ReduceTest {
+
+  private static final String inputTopic = "inputTopic";
+  private static final String outputTopic = "outputTopic";
+
+  @Test
+  public void shouldConcatenateWithReduce() {
+    final List<KeyValue<Integer, String>> inputRecords = Arrays.asList(
+        new KeyValue<>(456, "stream"),
+        new KeyValue<>(123, "hello"),
+        new KeyValue<>(123, "world"),
+        new KeyValue<>(456, "all"),
+        new KeyValue<>(123, "kafka"),
+        new KeyValue<>(456, "the"),
+        new KeyValue<>(456, "things"),
+        new KeyValue<>(123, "streams")
+    );
+
+    final Map<Integer, String> expectedOutput = new HashMap<>();
+    expectedOutput.put(456, "stream all the things");
+    expectedOutput.put(123, "hello world kafka streams");
+
+    // Step 1: Create the topology and its configuration
+    final StreamsBuilder builder = createTopology();
+    final Properties streamsConfiguration = createTopologyConfiguration();
+
+    try (final TopologyTestDriver topologyTestDriver = new TopologyTestDriver(builder.build(), streamsConfiguration)) {
+      // Step 2: Write the input
+      IntegrationTestUtils.produceKeyValuesSynchronously(
+          inputTopic, inputRecords, topologyTestDriver, new IntegerSerializer(), new StringSerializer());
+
+      // Step 3: Validate the output
+      final Map<Integer, String> actualOutput = IntegrationTestUtils.drainTableOutput(
+          outputTopic, topologyTestDriver, new IntegerDeserializer(), new StringDeserializer());
+      assertThat(actualOutput).hasSameSizeAs(expectedOutput);
+      assertThat(actualOutput).containsAllEntriesOf(expectedOutput);
+    }
+  }
+
+  private StreamsBuilder createTopology() {
+    final StreamsBuilder builder = new StreamsBuilder();
+    final KStream<Integer, String> input = builder.stream(inputTopic, Consumed.with(Serdes.Integer(), Serdes.String()));
+    final KTable<Integer, String> concatenated =
+        input.groupByKey(Grouped.with(Serdes.Integer(), Serdes.String())).reduce((v1, v2) -> v1 + " " + v2);
+    concatenated.toStream().to(outputTopic, Produced.with(Serdes.Integer(), Serdes.String()));
+    return builder;
+  }
+
+  private Properties createTopologyConfiguration() {
+    final Properties streamsConfiguration = new Properties();
+    streamsConfiguration.put(StreamsConfig.APPLICATION_ID_CONFIG, "reduce-test");
+    streamsConfiguration.put(StreamsConfig.BOOTSTRAP_SERVERS_CONFIG, "dummy config");
+    streamsConfiguration.put(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, "earliest");
```

Comment:
Also here, this config shouldn't be necessary.

Reply:
Ack